File: front/src/app/core/model/Especialidade.ts
export default class Especialidade {
  private _id: number;
  private _descricao: string;
  private _valorConsulta: number;
  private _limiteDiario: number;


  constructor(id: number, descricao: string, valorConsulta: number, limiteDiario: number) {
    this._id = id;
    this._descricao = descricao;
    this._valorConsulta = valorConsulta;
    this._limiteDiario = limiteDiario;
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get descricao(): string {
    return this._descricao;
  }

  set descricao(value: string) {
    this._descricao = value;
  }

  get valorConsulta(): number {
    return this._valorConsulta;
  }

  set valorConsulta(value: number) {
    this._valorConsulta = value;
  }

  get limiteDiario(): number {
    return this._limiteDiario;
  }

  set limiteDiario(value: number) {
    this._limiteDiario = value;
  }
}

File: back/app/models/Especialidade.py

class Especialidade:

    def __init__(self, codigo_especialidade: int, descricao: str, valor_consulta: float, limite_diario: int):
        self.codigo_especialidade = codigo_especialidade
        self.descricao = descricao
        self.valor_consulta = valor_consulta
        self.limite_diario = limite_diario

    def __str__(self):
        return (f"Cód Especialidade: {self.codigo_especialidade}, Descrição: {self.descricao}, "
                f"Valor: R$ {self.valor_consulta:.2f}, Limite Diário: {self.limite_diario}")

    def to_dict(self):
        return {
            "codigo_especialidade": self.codigo_especialidade,
            "descricao": self.descricao,
            "valor_consulta": self.valor_consulta,
            "limite_diario": self.limite_diario
        }
